evalml/automl/engine/engine_base.py

```diff
@@ -7,177 +7,204 @@
 from evalml.exceptions import PipelineScoreError
 from evalml.model_family import ModelFamily
 from evalml.preprocessing import split_data
 from evalml.problem_types import is_binary, is_multiclass
-from evalml.utils.logger import get_logger
 from evalml.utils.woodwork_utils import _convert_woodwork_types_wrapper
 
-logger = get_logger(__file__)
 
+class EngineComputation(ABC):
+    """Wrapper around the result of a (possibly asynchronous) engine computation."""
 
-class EngineBase(ABC):
-    """Base class for the engine API which handles the fitting and evaluation of pipelines during AutoML."""
-
-    def __init__(self, X_train=None, y_train=None, ensembling_indices=None, automl=None, should_continue_callback=None, pre_evaluation_callback=None, post_evaluation_callback=None):
-        """Base class for the engine API which handles the fitting and evaluation of pipelines during AutoML.
-
-        Arguments:
-            X_train (ww.DataTable): Training features
-            y_train (ww.DataColumn): Training target
-            ensembling_indices (list): Ensembling indices for ensembling data
-            automl (AutoMLSearch): A reference to the AutoML search. Used to access configuration and by the error callback.
-            should_continue_callback (function): Returns True if another pipeline from the list should be evaluated, False otherwise.
-            pre_evaluation_callback (function): Optional callback invoked before pipeline evaluation.
-            post_evaluation_callback (function): Optional callback invoked after pipeline evaluation, with args pipeline and evaluation results. Expected to return a list of pipeline IDs corresponding to each pipeline evaluation.
+    @abstractmethod
+    def get_result(self):
+        """Get the computation result.
+        Will block until the computation is finished.
+
+        Raises Exception: If computation fails. Returns traceback.
         """
-        self.X_train = X_train
-        self.y_train = y_train
-        self.automl = automl
-        self._should_continue_callback = should_continue_callback
-        self._pre_evaluation_callback = pre_evaluation_callback
-        self._post_evaluation_callback = post_evaluation_callback
-        self.ensembling_indices = ensembling_indices
 
     @abstractmethod
-    def evaluate_batch(self, pipelines):
-        """Evaluate a batch of pipelines using the current dataset and AutoML state.
+    def done(self) -> bool:
+        """Is the computation done?"""
 
-        Arguments:
-            pipeline_batch (list(PipelineBase)): A batch of pipelines to be fitted and evaluated
+    @abstractmethod
+    def cancel(self) -> None:
+        """Cancel the computation."""
 
-        Returns:
-            list (int): A list of the new pipeline IDs which were created by the AutoML search.
-        """
 
-    @abstractmethod
-    def train_batch(self, pipelines):
-        """Train a batch of pipelines using the current dataset.
+class JobLogger:
```

Comment:
I think it can go anywhere! But since we only need it for one pipeline job (that's also defined in engine base) I figured I'd put it here as opposed to its own file.